astroid/bases.py

```diff
@@ -104,13 +104,14 @@ def infer(self, context=None):
         yield self
 
 
-def _infer_stmts(stmts, context, frame=None):
+def _infer_stmts(stmts, context, frame=None, copy=True):
```

Comment:
@nickdrozd I feel the whole part with the copying needs to be treated at the context level. We would need a way to say that a context is fresh or not, and we need to use that instead of passing the flag down the line.